uni/lib/controller/parsers/parser_schedule.dart

```diff
@@ -34,12 +36,23 @@ Future<List<Lecture>> parseSchedule(http.Response response) async {
     final String classNumber = lecture['turma_sigla'];
     final int occurrId = lecture['ocorrencia_id'];
 
-    lectures.add(Lecture.fromApi(subject, typeClass, day, secBegin, blocks,
-        room, teacher, classNumber, occurrId));
+    DateTime monday = DateTime.now();
+    monday = DateUtils.dateOnly(monday);
+    //get closest monday
+    if(monday.weekday >=1 && monday.weekday <= 5){
+      monday = monday.subtract(Duration(days:monday.weekday-1));
+    } else {
+      monday = monday.add(Duration(days: DateTime.daysPerWeek - monday.weekday + 1));
+    }
```

Comment:
This should be in a util file, possibly extending DateTime

Reply:
Done